### ISSUE TITLE
AttributeError: 'str' object has no attribute 'decode'

### ISSUE THREAD

**Issue by cpremkumar** [toutpt/captionstransformer#1]
Hi,

I'm trying to use the captionstransformer as I've a need to convert SRT to TTML. I'm using the following example code but I'm getting the following error,

Error trace:
Traceback (most recent call last):
  File "srt_to_ttml.py", line 15, in <module>
    captions = reader.read()
  File "C:\Python34\lib\site-packages\captionstransformer-1.2.1-py3.4.egg\captionstransformer\core.py", line 13, in read
    self.rawcontent = self.rawcontent.decode(self.encoding)
AttributeError: 'str' object has no attribute 'decode'

The following is the example code that I'm using,
from captionstransformer.srt import Reader
from captionstransformer.ttml import Writer
from io import StringIO
test_content = StringIO(u"""
1
00:00:03,490 --> 00:00:07,430
FISHER: All right. So, let's begin.
This session is: Going Social

00:00:07,430 --> 00:00:11,600
with the YouTube APIs. I am
Jeff Fisher,
""")
reader = Reader(test_content)
captions = reader.read()
# len(captions) == 4
# first = captions[0]
# type(first.text) == unicode
# first.text == u"Jellyfish at the Monterey Aquarium"
# next get a writer

filelike = StringIO()
writer = Writer(filelike)
writer.set_captions(captions)
text = writer.captions_to_text()
text.startswith(u"""<tt xml:lang="" xmlns="http://www.w3.org/ns/ttml"><body><div>""")
writer.write()
writer.close()

Could you please help me fix this error? I'm using ActiveState Python 3.4.1

Thanks,
Prem

**Comment by medeaoblongata** [comment]
This is a python 2 vs 3 issue. You mention that you are using Python 3.4.1. This library is made for Python 2, but it is very easy to fix. (I hope toutpt will read this and post an update).

Python 3 **always** uses unicode for the str type, and the str type no longer has (or needs) a decode method. Reading a str from a file in python 3 will always give you unicode, which needs no further decoding. If you try to call str.decode() in python 3 you will get this error.

A quick-and-dirty solution to this is to change core.py in this library, so that decode **only** gets called if it exists (which is effectively the same as checking whether you are running python 2).  You could also use a try block, to catch that specific exception.

The decode method is called in **two** places in core.py. 

I'm providing some code compatible with both python2+3 that you can use:

...in the read() method of the Reader class. 
```
    def read(self):
        self.rawcontent = self.fileobject.read()        
        if type(self.rawcontent) == str:
            if hasattr(self.rawcontent, "decode"):
                self.rawcontent = self.rawcontent.decode(self.encoding)            
        self.text_to_captions()
        return self.captions
```
...and in the set_text() method of the Caption class
```
    def set_text(self, value):
        if type(value) == str:
            if hasattr(str, "decode"):
                value = str.decode(self.encoding)
        elif type(value) != unicode:
            raise ValueError("text must be either encoded string or unicode")
        self._text = value
```

This isn't a completely bulletproof solution, but in the absence of malicious code, it should 'just work'. I am sure better solutions can be imagined, but in my case, I just want to get it working.

BTW: Switching the decode issue will make the captionstransformer library compatible with python 2 & 3.

**Comment by toutpt** [comment]
closed by PR #2
